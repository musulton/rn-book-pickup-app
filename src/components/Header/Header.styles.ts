import {StyleSheet, TextStyle, ViewStyle} from "react-native";

import {COLORS, SIZES} from "../../constants";

interface Styles {
    container: ViewStyle
    username: TextStyle
    title: TextStyle
    input: TextStyle
    searchBar: ViewStyle
}

const styles = StyleSheet.create<Styles>({
    container: {
        display: "flex",
        justifyContent: "flex-start",
        padding: SIZES.padding1,
        borderBottomWidth: 1,
        borderBottomColor: COLORS.lightGray2,
    },
    username: {
        color: COLORS.lightGray
    },
    title: {
        fontSize: 20,
        fontWeight: "bold",
    },
    input: {
        fontSize: 20,
        marginLeft: SIZES.padding2,
        width: "90%",
    },
    searchBar: {
        marginTop: SIZES.padding2,
        padding: SIZES.padding2,
        flexDirection: "row",
        backgroundColor: COLORS.lightGray2,
        borderRadius: SIZES.borderRadius,
        alignItems: "center",
    }
})


export default styles
