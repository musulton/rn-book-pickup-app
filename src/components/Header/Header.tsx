import React from "react";
import {Text, TextInput, View} from "react-native";

import styles from "./Header.styles";
import Icon from "../Icon";

const Header = () => {
    const [clicked, setClicked] = React.useState(false);
    const [value, setValue] = React.useState("")

    return (
        <View style={styles.container}>
            <Text style={styles.username}>Hi, Your account name</Text>
            <Text style={styles.title}>Discover Latest Book</Text>
            <View style={styles.searchBar}>
                <TextInput
                    style={styles.input}
                    placeholder="Search book.."
                    value={value}
                    onChangeText={setValue}
                    onFocus={() => setClicked(true)}
                    onBlur={() => setClicked(false)}
                />
                {clicked && (
                    <Icon
                        name="close"
                        size={20}
                        onPress={() => setValue("")}
                    />
                )}
            </View>
        </View>
    )
}

export default Header
