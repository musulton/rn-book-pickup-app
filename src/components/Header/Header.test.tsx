import {fireEvent, render, screen, waitFor} from "@testing-library/react-native";

import Header from "./Header";

describe("<Header />", () => {
    it("should render correctly", () => {
        render(<Header />)

        expect(screen.toJSON()).toMatchSnapshot();
    })

    it("should render correctly when <SearchBook /> is filled", async () => {
        const expectedValue = "book title"

        render(<Header />)

        const searchInput = await screen.findByPlaceholderText("Search book..")

        fireEvent.press(searchInput)
        fireEvent.changeText(searchInput, expectedValue)

        await waitFor(() => {
            expect(screen.toJSON()).toMatchSnapshot()
        })
    })
});
