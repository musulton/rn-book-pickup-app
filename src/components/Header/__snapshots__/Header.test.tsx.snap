// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render correctly 1`] = `
<View
  style={
    {
      "borderBottomColor": "#EFEFF0",
      "borderBottomWidth": 1,
      "display": "flex",
      "justifyContent": "flex-start",
      "padding": 20,
    }
  }
>
  <Text
    style={
      {
        "color": "#64676D",
      }
    }
  >
    Hi, Your account name
  </Text>
  <Text
    style={
      {
        "fontSize": 20,
        "fontWeight": "bold",
      }
    }
  >
    Discover Latest Book
  </Text>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#EFEFF0",
        "borderRadius": 6,
        "flexDirection": "row",
        "marginTop": 10,
        "padding": 10,
      }
    }
  >
    <TextInput
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholder="Search book.."
      style={
        {
          "fontSize": 20,
          "marginLeft": 10,
          "width": "90%",
        }
      }
      value=""
    />
  </View>
</View>
`;

exports[`<Header /> should render correctly when <SearchBook /> is filled 1`] = `
<View
  style={
    {
      "borderBottomColor": "#EFEFF0",
      "borderBottomWidth": 1,
      "display": "flex",
      "justifyContent": "flex-start",
      "padding": 20,
    }
  }
>
  <Text
    style={
      {
        "color": "#64676D",
      }
    }
  >
    Hi, Your account name
  </Text>
  <Text
    style={
      {
        "fontSize": 20,
        "fontWeight": "bold",
      }
    }
  >
    Discover Latest Book
  </Text>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#EFEFF0",
        "borderRadius": 6,
        "flexDirection": "row",
        "marginTop": 10,
        "padding": 10,
      }
    }
  >
    <TextInput
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholder="Search book.."
      style={
        {
          "fontSize": 20,
          "marginLeft": 10,
          "width": "90%",
        }
      }
      value="book title"
    />
  </View>
</View>
`;
